client/my-sites/stats/stats-list/action-follow.jsx

```diff
@@ -2,61 +2,84 @@
  * External dependencies
  */
 import React from 'react';
-import createReactClass from 'create-react-class';
-import { localize } from 'i18n-calypso';
 import classNames from 'classnames';
-import debugFactory from 'debug';
+import config from '@automattic/calypso-config';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-/* eslint-disable no-restricted-imports */
-import observe from 'calypso/lib/mixins/data-observe';
-/* eslint-enable no-restricted-imports */
-import { gaRecordEvent } from 'calypso/lib/analytics/ga';
 import Gridicon from 'calypso/components/gridicon';
+import wpcom from 'calypso/lib/wp';
+import { gaRecordEvent } from 'calypso/lib/analytics/ga';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { getSiteDomain } from 'calypso/state/sites/selectors';
+import { requestSiteStats } from 'calypso/state/stats/lists/actions';
 
-const debug = debugFactory( 'calypso:stats:action-follow' );
-
-const StatsActionFollow = createReactClass( {
-	displayName: 'StatsActionFollow',
+class StatsActionFollow extends React.Component {
+	state = {
+		isFollowing: this.props.isFollowing,
+	};
 
-	mixins: [ observe( 'followSite' ) ],
+	refreshFollowers = () => {
+		this.props.requestSiteStats( this.props.selectedSiteId, 'statsFollowers', {
+			type: 'wpcom',
+			max: 10,
+		} );
+		this.props.requestSiteStats( this.props.selectedSiteId, 'statsCommentFollowers', { max: 7 } );
+	};
 
-	clickHandler: function ( event ) {
-		const site = this.props.followSite;
+	clickHandler = ( event ) => {
 		let gaEvent;
 
 		event.stopPropagation();
 		event.preventDefault();
-		debug( 'handling follower click', this.props.data );
 
-		if ( ! site.is_following ) {
+		if ( ! this.state.isFollowing ) {
 			gaEvent = 'Follow';
-			site.follow();
+			wpcom
+				.site( this.props.siteId )
+				.follow()
+				.add( { source: config( 'readerFollowingSource' ) } )
+				.then( this.refreshFollowers );
+
+			// Intentionally optimistic update.
+			this.setState( {
+				isFollowing: true,
+			} );
 		} else {
 			gaEvent = 'Unfollow';
-			site.unfollow();
+			wpcom
+				.site( this.props.siteId )
+				.follow()
+				.del( { source: config( 'readerFollowingSource' ) } )
+				.then( this.refreshFollowers );
```

Comment:
I guess refetching the whole list was intentional but it differs in a significant way from prod where it blinks/flashes when new data is fetched. Does it make sense to do that update silently?

Reply:
I gave this a bit more thought and actually, we don't need to request all those stats just for the "is following" flag. Now that we handle errors, we should be good to rely on component state. Changed to be silent as of https://github.com/Automattic/wp-calypso/pull/49825/commits/d8fe8edca45765bdc05221c7cdddd93c014ccb11